ocs_archive/input/file.py

```diff
@@ -124,16 +159,26 @@ def _repair_public_date(self):
         # Set the public date based on observation date. Should be overriden if you have another method
         # of specifying the public date in your file type
         if not self.header_data.get_public_date():
-            if (self.header_data.get_configuration_type() in settings.CALIBRATION_TYPES or
-                    (self.header_data.get_proposal_id() and any([prop in self.header_data.get_proposal_id() for prop in settings.PUBLIC_PROPOSALS]))):
-                public_date = self.header_data.get_observation_date()
-            elif ((self.header_data.get_proposal_id() and any([prop in self.header_data.get_proposal_id() for prop in settings.PRIVATE_PROPOSALS])) or
-                    any([chars in self.open_file.basename for chars in settings.PRIVATE_FILE_TYPES])):
-                # This should be private, set it to 999 years from observation date
-                public_date = (parse(self.header_data.get_observation_date()) + timedelta(days=365 * 999)).isoformat()
+            # Check if the proposal is denoted as public or private in the observation portal
+            if self.data_privacy_tags:
+                if any([tag in settings.PRIVATE_PROPOSAL_TAGS for tag in self.proposal_tags]):
```

Comment:
We already know which privacy tag this has from the `self.data_privacy_tags`, so can't we just use that here? Like if any([tag in settings.PRIVATE_PROPOSAL_TAGS for tag in self.data_privacy_tags]) and then else if for the public condition and then else for the fallback

Reply:
Ah right - meant to clean that up. Will do.